### PR TITLE
ci: Check if valid title is used in PR

### LINKED ISSUE #814: Reduce release_notes.md merge conflicts
Now that Flank is actively developed, there are a lot of merge conflicts in release_notes.md. Maybe we should have a bunch of empty lines to reduce merge conflicts?

> Conflicting files
>
> release_notes.md

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #814

- There is additional trigger which checks PR title
- PR generate release notes and append to `release_notes.md`

## Checklist

- [x] Documented
- [x] Unit tested
- [x] release_notes.md updated

### CODE REVIEW

**Comment by bootstraponline** [comment]
> .github/workflows/after_pr.yml

What are the tradeoffs of generating release notes after PRs are merged vs before they're merged? It might be nice to review the generated release notes as part of the pull request.

**Comment by piotradamczyk5** [comment]
> > .github/workflows/after_pr.yml
> 
> What are the tradeoffs of generating release notes after PRs are merged vs before they're merged? It might be nice to review the generated release notes as part of the pull request.

It generates release notes based on PR title **after** merging, if we do it before, then we will have merge conflicts anyway.
If you would like to check generated release notes just check the title of PR 😃

**Comment by bootstraponline** [comment]
that makes sense! I'm looking into creating a Flank app and then adding the token from that.

**Comment by bootstraponline** [comment]
> Bot accounts are specific to GitHub Apps and are built into every GitHub App.
>
> https://docs.github.com/en/developers/apps/differences-between-github-apps-and-oauth-apps

There's some way to get a bot account from a GitHub App and have the access token associated with the app. I think we'll need to figure this out. We're trying to move away from user accounts/personal access tokens.

It might make sense to fork flank into a new org, experiment there, and then we'll bring back the learnings to the production Flank repo.

**Comment by bootstraponline** [comment]
https://docs.github.com/en/developers/apps/authenticating-with-github-apps#http-based-git-access-by-an-installation

I created https://github.com/apps/flank but am not sure where the token is. 🤔

**Comment by piotradamczyk5** [comment]
> https://docs.github.com/en/developers/apps/authenticating-with-github-apps#http-based-git-access-by-an-installation
> 
> I created https://github.com/apps/flank but am not sure where the token is. 🤔

Maybe [here](https://developer.github.com/v3/apps/#create-an-installation-access-token-for-an-app) is the answer ?
I do not know how it works

**Comment by bootstraponline** [comment]
I'm super excited for the conventional commits automatic release notes generation. Thanks for helping me think through a few different possibilities. If we generate the release notes after every commit, that'll add a lot of noise in the git history of the project.

After reflecting, I think the best way to generate release notes is directly before a release. This matches the angular convention where this practice originated from.

https://github.com/angular/angular/commits/master/CHANGELOG.md

This also resolves the problem of having to provide a highly privileged token that interfaces with a protected branch. What do you think?

**Comment by piotradamczyk5** [comment]
> https://github.com/angular/angular/commits/master/CHANGELOG.md



> I'm super excited for the conventional commits automatic release notes generation. Thanks for helping me think through a few different possibilities. If we generate the release notes after every commit, that'll add a lot of noise in the git history of the project.
> 
> After reflecting, I think the best way to generate release notes is directly before a release. This matches the angular convention where this practice originated from.
> 
> https://github.com/angular/angular/commits/master/CHANGELOG.md
> 
> This also resolves the problem of having to provide a highly privileged token that interfaces with a protected branch. What do you think?

Ok I will split this PR into 2 separate.
- This will just check if conventional commit title is right
- second one I will change a logic to generating release notes from last release to current one and make some PR with them

Is it ok?

**Comment by bootstraponline** [comment]
Sounds good!

File: release_notes.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 - [#987](https://github.com/Flank/flank/pull/987) Flank Error Monitoring readme addition ([sloox](https://github.com/Sloox))
 - [#990](https://github.com/Flank/flank/pull/990) Fix: exclusion of @Suppress test. ([piotradamczyk5](https://github.com/piotradamczyk5))
 - [#988](https://github.com/Flank/flank/pull/988) Add versions description command for ios and android. ([adamfilipow92](https://github.com/adamfilipow92))
+- [#948](https://github.com/Flank/flank/pull/948) CI changes: Check if valid title is used in PR. ([piotradamczyk5](https://github.com/piotradamczyk5))
```
Comment (adamfilipow92) [inline release_notes.md:6]: Maybe its should be
[#989](https://github.com/Flank/flank/pull/989)
?

Comment (piotradamczyk5) [reply release_notes.md]: thanks 👍

**Pull request merged**